addons/crm/wizard/crm_lead_to_opportunity.py

```diff
@@ -158,7 +161,10 @@ def _convert_and_allocate(self, leads, user_ids, team_id=False):
 
     def _convert_handle_partner(self, lead, action, partner_id):
         # used to propagate user_id (salesman) on created partners during conversion
+        should_create = action == 'create'
         lead.with_context(default_user_id=self.user_id.id)._handle_partner_assignment(
             force_partner_id=partner_id,
-            create_missing=(action == 'create')
+            create_missing=should_create
         )
+        if self.company_partner_id and should_create:
```

Comment:
Feels like this should be propagated to the methods and handled at partner creation time